libraries/bsg_manycore_features.h

```diff
@@ -29,9 +29,9 @@
 #define BSG_MANYCORE_FEATURES_H
 // <features.h> sorts out many of these defines based on compile time flags (e.g. -std=c++11)
 #include <features.h>
-// check _BSG_SOURCE
-#ifndef _BSD_SOURCE
-#error "_BSD_SOURCE not defined: required for bsg_manycore_runtime"
+// check _DEFAULT_SOURCE
```

Comment:
IIRC _DEFAULT_SOURCE and _BSD_SOURCE should both be defined

Reply:
Good catch! Yeah we need both to support both `glibc 2.17 (centos 7)` and `glibc 2.28 (rocky 8)`. The `glibc` version can be checked with `ldd --version` command.

I think a more appropriate fix is to add -D_BSD_SOURCE to `libraries/platforms/common/dpi/library.mk`, which is the root cause of this error, @dpetrisko what do you think?